### PR TITLE
Add test cases to improve mutation score indicator

### PR DESCRIPTION
**Pull request by Kourin1996**
# Description

Add test cases in message package to improve MSI (mutation score indicator)

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have added sufficient documentation in code

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: messages/event_manager.go [hunk 1]
```diff
@@ -99,8 +99,9 @@ func (em *eventManager) close() {
 	em.subscriptionsLock.Lock()
 	defer em.subscriptionsLock.Unlock()
 
-	for _, subscription := range em.subscriptions {
```
Comment (JekaMas) [inline messages/event_manager.go:102]: Could you also take care about https://github.com/0xPolygon/go-ibft/pull/38 ? (merge main, finish task)

File: messages/event_manager_test.go [hunk 2]
```diff
@@ -1,26 +1,212 @@
...
+		delete(em.subscriptions, sub.ID)
+	}
+
+	t.Run("should exit before locking subscriptionsLock if numSubscriptions is zero", func(t *testing.T) {
+		t.Parallel()
+
+		em, sub := setupEventManagerAndSubscription(t)
+
+		// overwrite numSubscription
+		atomic.StoreInt64(&em.numSubscriptions, 0)
```
Comment (JekaMas) [inline messages/event_manager_test.go:117]: Why do we need it? Could we do the same inside the guarded area(the lock one line below)?

**Comment by JekaMas** [comment]
What is the result score?

**Comment by Kourin1996** [comment]
@JekaMas thanks for reviewing. `make mut` told the score was 0.989011. But this score is sometimes different every time I ran mutation test, I will run mutation test 10 times. And will finish fix of lint issue today.

**Comment by Kourin1996** [comment]
@JekaMas I have one small question. I've not seen any issue from `core` package. Does it mean there is no issue in `core` package or we're skipping test for `core` for now?

File: messages/event_manager_test.go (hunk 2 above)
Comment (Kourin1996) [reply messages/event_manager_test.go:123]: This is for safe check to make sure early return occurs if numSubscriptions is zero

**Review by JekaMas — APPROVED** [review]

File: messages/event_manager.go (hunk 1 above)
Comment (Kourin1996) [reply messages/event_manager.go:102]: I merged latest `develop` and fix conflict

**Review by b00ris — APPROVED** [review]

File: .golangci.yml [hunk 3]
```diff
@@ -310,7 +308,7 @@ linters-settings:
       # https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#function-length
       - name: function-length
         severity: warning
-        disabled: false
+        disabled: true
```
Comment (JekaMas) [inline .golangci.yml:311]: is it relevant to the PR?

**Review by yaziciahmet — APPROVED** [review]

File: .golangci.yml (hunk 3 above)
Comment (Kourin1996) [reply .golangci.yml:311]: I think this is by the change of latest develop cause I didn't change this file and just merged latest

**Pull request merged**